src/plugins/data_view_editor/public/components/confirm_modals/edit_data_view_changed_modal.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useState } from 'react';
+import { EuiCallOut, EuiSpacer, EuiConfirmModal, EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+const geti18nTexts = (dataViewName?: string) => ({
+  cancelButtonText: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.cancelButtonLabel',
+    {
+      defaultMessage: 'Cancel',
+    }
+  ),
+  confirmButtonText: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.saveButtonLabel',
+    {
+      defaultMessage: 'Edit',
+    }
+  ),
+  warningEditingDataView: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.warningEditingDataView',
+    {
+      defaultMessage:
+        'Are you sure you want to edit this data view? Changing it might have unwanted outcomes.',
```

Comment:
>Changing it might have unwanted outcomes

IMO this is too vague. Might also be good to get @gchaps to look at this pr. 'Anything that depends upon this might break.' - first swing at better text. 


Also, this dialog is displayed when only the name is changed. Nothing will break in this circumstance.